apps/language_server/lib/language_server/providers/workspace_symbols.ex

```diff
@@ -477,7 +477,16 @@ defmodule ElixirLS.LanguageServer.Providers.WorkspaceSymbols do
     end)
   end
 
-  @spec build_result(key_t, symbol_t, String.t(), nil | non_neg_integer) :: symbol_information_t
+  @spec build_result(
+          key_t,
+          symbol_t,
+          String.t(),
+          nil | non_neg_integer | {non_neg_integer, non_neg_integer}
+        ) :: symbol_information_t
+  defp build_result(key, symbol, path, {line, _}) do
+    build_result(key, symbol, path, line)
+  end
+
```

Comment:
Please add a TODO remove when we require OTP 24

Reply:
I don't think it should be removed. Both patterns should be supported if we want to support OTP 24 and earlier versions.

Reply:
In a few years eventually support for OTP < 24 will be dropped. By that time no-one will remember why there are two definitions. That's why we need a TODO

Reply:
OTP 24 also support both [patterns](https://erlang.org/doc/man/erl_anno.html#type-location) depending on `ERL_COMPILER_OPTIONS` (see https://github.com/erlang/otp/pull/3006). I've moved the pattern matching to `build_range/1`, created a typedef `erl_location_t` and renamed `line` variables to `location`, it seems a bit cleaner.